### PR TITLE
suitesparse, luajit, lua-lgi: Update them

### PR DESCRIPTION
**Pull request by brunvonlope**

### CODE REVIEW

**Comment by brunvonlope** [comment]
Testing as requested in: https://github.com/DrTimothyAldenDavis/SuiteSparse/issues/862#issuecomment-2282860977

**Comment by brunvonlope** [comment]
And... failed. @hfiguiere Do you know if this restriction is due to security reasons?

File: org.gimp.GIMP.json [hunk 1]
```diff
@@ -645,15 +645,9 @@
...
-                    "sha256": "1ad2e34b111c802f9d0cdf019e986909123237a28c746b21295b63c9e785d9c3",
-                    "x-checker-data": {
-                        "type": "anitya",
-                        "project-id": 6444,
-                        "stable-only": false,
-                        "url-template": "https://luajit.org/download/LuaJIT-$version.tar.gz"
-                    }
+                    "type": "git",
+                    "url": "https://github.com/LuaJIT/LuaJIT.git",
+                    "branch": "v2.1"
```
Comment (hfiguiere) [inline org.gimp.GIMP.json:650]: you NEVER build a branch. This break reproducibility.

File: org.gimp.GIMP.json [hunk 2]
```diff
@@ -680,8 +671,8 @@
             "sources": [
                 {
                     "type": "git",
-                    "url": "https://github.com/pavouk/lgi.git",
-                    "commit": "c21f35fccae87fd4e3625d4c878f584b7255d6f6"
+                    "url": "https://github.com/lgi-devs/lgi.git",
+                    "branch": "master"
```
Comment (hfiguiere) [inline org.gimp.GIMP.json:675]: same here.

File: org.gimp.GIMP.json (hunk 1 above)
Comment (brunvonlope) [reply org.gimp.GIMP.json:650]: That's what I was questioning in https://github.com/flathub/org.gimp.GIMP/pull/344#issuecomment-2295348346. Thanks for the response.

**Comment by brunvonlope** [comment]
bot, build

**Pull request merged**